scaladoc/src/dotty/tools/scaladoc/site/StaticSiteLoader.scala

```diff
@@ -69,15 +70,16 @@ class StaticSiteLoader(val root: File, val args: Scaladoc.Args)(using StaticSite
           nested.map(child => loadChild(categoryPath)(child))
         ) { indexPath =>
           val indexPathDirectory = Paths.get(indexPath).getParent
-          val filesInDirectory = Option(root.toPath.resolve(indexPathDirectory).toFile.listFiles)
+          val filesInDirectory = Option(ctx.docsPath.resolve(indexPathDirectory).toFile.listFiles)
           filesInDirectory.fold(List.empty) { files =>
             val mappingFunc: File => File = file => {
-              val relativeFile = root.toPath.resolve(indexPathDirectory).relativize(file.toPath)
+              val relativeFile = ctx.docsPath.resolve(indexPathDirectory).relativize(file.toPath)
               categoryPath.resolve(relativeFile).toFile
             }
             files.toList
               .filter(_.toPath != indexPage.file.toPath)
               .flatMap(file => loadRecursively(file, mappingFunc))
+              .sorted
```

Comment:
I am not sure that sorting pages alphabetically is a good default choice. It rarely matches what we want. See e.g. the Enums section in the reference documentation, [which now shows first the advanced “Algebraic Data Types” page before the “Enumerations” introduction page](https://scala3doc.virtuslab.com/pr-simplify-sidebar-config/scala3/docs/reference/enums/index.html). You can compare here with the current docs [here](https://docs.scala-lang.org/scala3/reference/enums/index.html).

Reply:
in the current docs.scala-lang the ordering of the page in a "collection" (directory in this case) is part of the file metadata, i.e. an int representing the page index - this seems better to me to avoid having to synchronise with an external sidebar file.

We could also add checks to ensure each page has a valid ordering - and perhaps it could be more flexible than a 1:1 with index, so that the ordering value just has to increase (allowing for more flexible 'buckets' for when we optionally add extra 'details'  pages to an existing document, without having to fix the indices of every other page). e.g. (`1`,`2`,`10`,`20`,`30`,`31`, `32`, `50` etc.)